modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,374 @@
+        
+        auto dimensions = List<IDimension>();
+        if (useExplicitDomain) 
+            dimensions.pushBack(Dimension(ListDimensionRule(explicitDimension)));
+        else 
+        {
+            auto inputLowValue = static_cast<Float>(inputDataDescriptor.getValueRange().getLowValue());
+            auto inputHighValue = static_cast<Float>(inputDataDescriptor.getValueRange().getHighValue());
+
+            size_t rangeSize = inputHighValue - inputLowValue;
+            rangeSize = (rangeSize / classCount) + (rangeSize % classCount != 0) + 1;
+            dimensions.pushBack(Dimension(LinearDimensionRule(classCount, (Int)inputLowValue, rangeSize)));
+        }
+        outputDataDescriptorBuilder.setDimensions(dimensions);
+
+        outputDataDescriptorBuilder.setValueRange(Range(0, 1));
+
+        if (outputName.empty())
+            outputDataDescriptorBuilder.setName(inputDataDescriptor.getName().toStdString() + "/Classified");
+        else
+            outputDataDescriptorBuilder.setName(outputName);
+
+        outputDataDescriptorBuilder.setUnit(Unit("%"));
+
+        outputDataDescriptor = outputDataDescriptorBuilder.build();
+        outputSignal.setDescriptor(outputDataDescriptor);
+        
+        outputDomainDataDescriptor = DataDescriptorBuilderCopy(inputDomainDataDescriptor).setRule(ExplicitDataRule()).build();
+        outputDomainSignal.setDescriptor(outputDomainDataDescriptor);
+    }
+    catch (const std::exception& e)
+    {
+        LOG_W("ClassifierFb: Failed to set descriptor for classification signal: {}", e.what())
+        outputSignal.setDescriptor(nullptr);
+    }
+}
+
+
+void ClassifierFbImpl::onPacketReceived(const InputPortPtr& port)
+{
+    std::scoped_lock lock(sync);
+
+    PacketPtr packet;
+    const auto connection = inputPort.getConnection();
+    if (!connection.assigned())
+        return;
+
+    packet = connection.dequeue();
+
+    while (packet.assigned())
+    {
+        switch (packet.getType())
+        {
+            case PacketType::Event:
+                processEventPacket(packet);
+                break;
+
+            case PacketType::Data:
+                SAMPLE_TYPE_DISPATCH(inputSampleType, processDataPacket, packet);
+                break;
+
+            default:
+                break;
+        }
+
+        packet = connection.dequeue();
+    };
+}
+
+void ClassifierFbImpl::processEventPacket(const EventPacketPtr& packet)
+{
+    if (packet.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
+    {
+        DataDescriptorPtr inputDataDescriptor = packet.getParameters().get(event_packet_param::DATA_DESCRIPTOR);
+        DataDescriptorPtr inputDomainDataDescriptor = packet.getParameters().get(event_packet_param::DOMAIN_DATA_DESCRIPTOR);
+        processSignalDescriptorChanged(inputDataDescriptor, inputDomainDataDescriptor);
+    }
+}
+
+inline UInt ClassifierFbImpl::timeMs(UInt time) 
+{
+    return time * inputDeltaTicks;
```

Comment:
Assuming that the input is in seconds, to convert to ms, the "ticks" (raw data of the domain signal) should be multiplied as follows:
```
time * inputDomainDescriptor.getTickResolution() * (1 / 1000)
```